clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fmap.py

```diff
@@ -245,165 +252,220 @@ def fmap_image(
+
+
+def check_case_fmap(fmap_path: Path) -> str or None:
+    sub = fmap_path.parent.parent.name
+    ses = fmap_path.parent.name
+
+    files = [f for f in os.listdir(fmap_path) if not f.startswith(".")]
+    nb_files = len(files)
+
+    extensions = [
+        re.search(r"fmap(.*).json", f).group(1)
+        for f in os.listdir(fmap_path)
+        if re.search(r"fmap(.*).json", f)
+    ]
+
+    if nb_files == 0:
+        cprint(f"Folder for {sub}, {ses} is empty", lvl="warning")
+        return None
+    elif nb_files == 4:
+        if set(extensions) == set([""]):
+            cprint(
+                f"BIDS Case 3: expecting 1 magnitude and 1 fieldmap files for {sub}, {ses}.",
+                lvl="info",
+            )
+            return "case3"
         else:
-            if filename[:-12] == "fmap_ph.json":
-                phase_json = os.path.join(subroot, filename[:-12] + "phase1.json")
-                os.rename(fmap_filename, phase_json)
-                if os.path.exists(nifti_file):
-                    phase_nifti = os.path.join(
-                        subroot, filename[:-12] + "phase1.nii.gz"
-                    )
-                    os.rename(nifti_file, phase_nifti)
-            if filename[:-15] == "fmap_e2_ph.json":
-                phase_json = os.path.join(subroot, filename[:-15] + "phase2.json")
-                os.rename(fmap_filename, phase_json)
-                if os.path.exists(nifti_file):
-                    phase_nifti = os.path.join(
-                        subroot, filename[:-15] + "phase2.nii.gz"
-                    )
-                    os.rename(nifti_file, phase_nifti)
+            cprint(
+                f"BIDS Case 1 : expecting 1 magnitude and 1 phase files for {sub}, {ses}.",
+                lvl="info",
+            )
+            return "case1"
+    elif nb_files == 6 and set(extensions) == set(["_e1", "_e2", "_e2_ph"]):
+        cprint(
+            f"BIDS Case 1 : expecting 1 phase and 2 magnitude files for {sub}, {ses}.",
+            lvl="info",
+        )
+        return "case1"
+    elif nb_files == 8 and set(extensions) == set(["_e1", "_e2", "_e1_ph", "_e2_ph"]):
+        cprint(
+            f"BIDS Case 2 : expecting 2 phase and 2 magnitude files for {sub}, {ses}.",
+            lvl="info",
+        )
+        return "case2"
+    else:
+        cprint(f"No BIDS case was recognized for {sub}, {ses}.", lvl="warning")
+        return "unrecognized"
+
+
+def reorganize_fmaps(bids_path: Path):
+    """
+    Performs the renaming and suppression of fmap files according to BIDS specifications.
+
+    Parameters
+    ----------
+    bids_path : Path to the BIDS dataset
+
+    """
+
+    bids_case = {
+        "case1": case1_1phase_2mag,
+        "case2": case2_2phase_2mag,
+        "case3": direct_fieldmap,
+        "unrecognized": unrecognized_fmap_case,
+    }
+
+    lst_subjects = [d for d in os.listdir(bids_path) if "sub-ADNI" in d]
+
+    for subject in lst_subjects:
```

Comment:
Since `bids_path` is a `Path` object, you can simplify these lines into:

```python
for subject in (folder for folder in bids_path.iterdir() if folder.name.startswith("sub-ADNI")):
```